src/main.rs

```diff
@@ -82,14 +82,16 @@ fn try_main() -> Result<(), ProgramError> {
         AutofixMode::None
     };
 
-    let mut only: Option<HashSet<String>> = matches.values_of("only").map(|values| values.map(|s| s.to_string()).collect());
+    let mut only: HashSet<String> = matches
+        .values_of("only")
+        .map_or(HashSet::new(), |values| {
```

Comment:
I'm curious, do you consider `map_or(fallback, mapper)` clearer than `map(mapper).or(fallback)` (or actually, `.or_else(|| fallback)`)?

Reply:
Or `map_or_else(|| fallback, mapper)` ;)

I have no opinion. clippy does. I just do what clippy says. If I have opinions I change clippy or complain in the issues and then change it or my opinion.

Reply:
Great opinion-finding algorithm.

I like the `.map.or` better because you can put it on multiple lines and each line does a specific thing, but at the same time don't care enough because _this_ `map_or` is pretty straightforward. I also don't care about using the `_else` variant here as an empty HashSet doesn't allocate.

Reply:
for this specific case, clippy should probably be suggesting `.map(m).unwrap_or_default()`